lib/spack/spack/package.py

```
@@ -176,39 +176,41 @@ class DetectablePackageMeta(object):
     for the detection function.
     """
     def __init__(cls, name, bases, attr_dict):
-        # If a package has the executables attribute then it's
+        # If a package has the executables or libraries  attribute then it's
         # assumed to be detectable
-        if hasattr(cls, 'executables'):
+        if hasattr(cls, 'executables') or hasattr(cls, 'libraries'):
```

Can you raise an exception if `hasattr(cls, 'executables') and hasattr(cls, 'libraries')`?